filebeat/prospector/prospector.go

```diff
@@ -36,11 +36,15 @@ type Prospectorer interface {
 	Run()
 }
 
-func NewProspector(cfg *common.Config, states file.States, spoolerChan chan *input.Event) (*Prospector, error) {
+type Outlet interface {
```

Comment:
well, it's specific to prospector/crawler as is. I'd keep it here as full name is `prospector.Outlet` overlapping interfaces can be defined anywhere in go.